docs/developer-guide/formatters.md

```diff
@@ -79,6 +79,23 @@ And the second argument (`returnValue`) is an object with one or more of the fol
 }
 ```
 
+## Passing arguments
+
+To pass arguments, you can:
+
+- use environmental variables
+- pipe the output from the JSON formatter into another program
```

Comment:
It's not clear for me how piping output is passing arguments to formatter. Piping happens after formatter was ran.

Reply:
You're right. It's not clear that the second approach is a separate formatting program rather than a stylelint formatter. I've updated the docs to hopefully illustrate that.

---

Example formatting program:

```
// my-program-that-reads-JSON.js
const getStdin = require("get-stdin");
(async () => {
  const jsonString = await getStdin();
  const object = JSON.parse(jsonString);
  let colors = false;
  if (process.argv[2] && process.argv[2] === "-c") colors = true;
  console.dir(object, { depth: null, colors });
})();
```

```
stylelint -f json "*.css" | my-program-that-reads-JSON --c
```

This just (optionally) colourises and indents the JSON, but a formatting program could do anything with the data.